src/core/StateManagerStart.js

```diff
@@ -10,52 +10,48 @@
         Slide = require('./Slide'),
         ContentSnapshot = require('./ContentSnapshot');
 
-    function cleanup(children) {
-        var i = 0,
-            l = children.length;
-
-        for(; i < l; i += 1) {
-            if (children[i] && (children[i] instanceof ContentSnapshot)) {
-                children[i].destroy();
-            }
-        }
-    }
-
     function StateManagerStart(stateId, slideOutOptions, slideInOptions) {
-        var _slide,
+        var _exitSlide,
             _introSlide,
+            _exitSnapshot,
             _stateManager = this,
             _state = _stateManager.states[stateId],
             _args = [].slice.call(arguments),
             _cachedStateCreate = _state.create;
 
-        _stateManager.game.stage && cleanup(_stateManager.game.stage.children);
-
-        if (_stateManager.game.isBooted && slideOutOptions) {
-            _slide = new Slide(this.game);
-
-            (function (_state, slideOutOptions, slideInOptions) {
-                _state.create = function () {
-                    _cachedStateCreate.call(this);
-
-                    // Slide in intro
-                    if (slideInOptions) {
-                        _introSlide = new Slide(_stateManager.game);
-                        _stateManager._created = false;
-                        _introSlide.go(slideInOptions);
-
-                        _introSlide._transition.onComplete = function () {
+        if (_stateManager.game.isBooted && (slideOutOptions || slideInOptions) ) {
+            //need to take a snapshot before the next state's create function, otherwise everything on screen will be wiped out and snapshot will be empty
+            if(slideOutOptions) _exitSnapshot = new ContentSnapshot(this.game);
```

Comment:
Just for keep it in style, please wrap it in brackets. And multiline with proper spacing.